### ISSUE TITLE
`useless_attribute` incorrectly triggers for unknown tool attributes on `use`

### ISSUE THREAD

**Issue by lcnr** [rust-lang/rust-clippy#13764]
### Description

https://github.com/rust-lang/rust/pull/133643 failed in CI due to an false positive of the `useless_attribute` lint.

https://github.com/rust-lang/rust/actions/runs/12088915947/job/33713323311

To repro, use the following inside of `rustc_middle`:
```rust
#[allow(rustc::non_glob_import_of_type_ir_inherent)]
use rustc_type_ir::inherent::SliceLike;
```
This results in
```
  error: useless lint attribute
    --> compiler/rustc_middle/src/ty/mod.rs:29:1
     |
  29 | #[allow(rustc::non_glob_import_of_type_ir_inherent)]
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: if you just forgot a `!`, use: `#![allow(rustc::non_glob_import_of_type_ir_inherent)]`
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_attribute
     = note: `-D clippy::useless-attribute` implied by `-D clippy::correctness`
     = help: to override `-D clippy::correctness` add `#[allow(clippy::useless_attribute)]`
```
I can imagine us adding an allow-list for lints in the rustc-namespace, however, I believe that in general, this lint should never trigger for lints from unknown namespaces.
